### DISCUSSION TITLE
Airflow Metrics not showing up in StatsD explorer

### DISCUSSION THREAD

**Discussion by DibyoGit** [apache/airflow#22902] [General]
### Apache Airflow version

2.1.0

### What happened

We are running the Airflow via docker-compose, connecting to a MySQL host. It hs Statd Exporter config in it but we can not see ny airflow metrics in the StatsD metrics

### What you think should happen instead

StatsD metrics should have shown airlfow scheduler, web metrics

### How to reproduce

Run docker compsoe fil by docker compsoe up and access statsd metrics in IP:9123 url and search for airflow metrics

### Operating System

Oracle Linux 7

### Deployment

Docker-Compose

### Deployment details

Sharing the docker-compose:

version: '3'
x-airflow-common:
  &airflow-common
  image: apache/airflow:2.1.0
  environment:
    - LOAD_EX=n
    - POSTGRES_USER=airflow
    - POSTGRES_PASSWORD=airflow
    - POSTGRES_DB=airflow
    - REDIS_HOST=redis
    - REDIS_PASSWORD=redis
    - AIRFLOW__SCHEDULER__STATSD_ON=True
    - AIRFLOW__SCHEDULER__STATSD_HOST=statsd-exporter
    - AIRFLOW__SCHEDULER__STATSD_PORT=8125
    - AIRFLOW__SCHEDULER__STATSD_PREFIX=airflow
    - AIRFLOW__CORE__SQL_ALCHEMY_CONN=mysql+mysqlconnector://mysqluserid:airflowpassword@10.0.0.0:3306/airflow_custmysql_db_name
    - AIRFLOW__CELERY__BROKER_URL=redis://:redis@redis:6379/1
    - AIRFLOW__CELERY__RESULT_BACKEND=db+postgresql://airflow:airflow@postgres/airflow
    - AIRFLOW__CORE__FERNET_KEY=pMrhjIcqUNHMYRk_ZOBmMptWR6o1DahCXCKn5lEMpzM=
    - AIRFLOW__CORE__EXECUTOR=CeleryExecutor
    - AIRFLOW_HOME=/opt/airflow
    - AIRFLOW_CONFIG=/opt/airflow/airflow.cfg
    - AIRFLOW__CORE__LOAD_EXAMPLES=False
    - AIRFLOW__CORE__LOAD_DEFAULT_CONNECTIONS=False
    - AIRFLOW__WEBSERVER__WORKERS=2
    - AIRFLOW__WEBSERVER__WORKER_REFRESH_INTERVAL=1800

services:
    airflow-init:
        << : *airflow-common
        container_name: airflow-init
        entrypoint: /bin/bash
        command:
        - -c
        - airflow users list || ( airflow db init &&
            airflow users create
            --role Admin
            --username admin
            --password password
            --email airflow@airflow.com
            --firstname first
            --lastname last )
        restart: on-failure

    webserver:
        << : *airflow-common
        container_name: airflow-webserver
        restart: always
        depends_on:
            - statsd-exporter
            - airflow-init
        ports:
            - "8080:8080"
        command: airflow webserver
        healthcheck:
            test: ["CMD-SHELL", "[ -f /opt/airflow/airflow-webserver.pid ]"]
            interval: 30s
            timeout: 30s
            retries: 3

    flower:
        << : *airflow-common
        container_name: airflow-flower
        restart: always
        depends_on:
            - statsd-exporter
            - airflow-init
        ports:
            - "5555:5555"
        command: airflow celery flower

    scheduler:
        << : *airflow-common
        container_name: airflow-scheduler
        restart: always
        depends_on:
            - webserver
            - airflow-init
        command: airflow scheduler

    worker:
        << : *airflow-common
        container_name: airflow-worker
        restart: always
        sysctls:
            - net.ipv4.tcp_keepalive_time=200
            - net.ipv4.tcp_keepalive_intvl=200
            - net.ipv4.tcp_keepalive_probes=5
        depends_on:
            - scheduler
            - airflow-init
        ports:
            - "8793"
        command: airflow celery worker

    statsd-exporter:
        image: prom/statsd-exporter
        container_name: airflow-statsd-exporter
        command: "--statsd.listen-udp=:8125 --web.listen-address=:9102"
        ports:
            - 9123:9102
            - 8125:8125/udp

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Reply by potiuk** [reply]
This is at most a discussion -  this is not an Airlfow Issue. Maybe someone who is familiar with both docker-compose and statsd will help you with configuring your deployment properly.

**Reply by DibyoGit** [reply]
This is just a trial with docker compose.. When we tried to deploy AIrlfow ith helm chart in Kubernetes and applied statsd config for port 8125 /9125 even then it was not possible,. We beleive somehow airlfow is not  sending its metrics to statsd exporter

**Reply by potiuk** [reply]
It's very likely - if it is not configured. The Official Helm Chart of Airflow has statsd embeddd and exporter is there. You can switch to it and it will work for you. This is the recommended solution that a) will work, b) will save the maintainers time from debugging your configuration issues. Debugging a configuration problem of every single person who wants to run their custom configuration and is not able to debug their configuration problems when existing offiicial Helm Chart with worktng statsd exporter is published by the maintainers, is a total waste of our time and does not scale.

With great powers, come great responsisbilitites. If you want to run Docker Compose in production or own, custom Helm Chart then you take on the responsibility of debugging it and solving the problems with configuration. We've made a lot of effort to debug and test the Helm Chart of ours to make it works. You either stand on the shoulders of those giants who did it, or decide to stand on your own. But don't ask the giants to do the heavy debugging lifting for you if you take another path.

My recommendation for this problem is - switch to the official Helm Chart of Airflow.
